engine/src/org/pentaho/di/job/entries/getpop/JobEntryGetPOP.java

```diff
@@ -716,8 +716,19 @@ public String getPassword() {
     return password;
   }
 
+  /**
+   * @return Returns resolved decrypted password or null
+   * in case of {@link #getPassword()} returns null.
+   */
   public String getRealPassword() {
-    return environmentSubstitute( getPassword() );
+    String resolvedPassword = environmentSubstitute( getPassword() );
+    if ( resolvedPassword != null ) {
```

Comment:
As noted below, this code should be in a Utils method ( core/org.pentaho.di.core.util.Utils I believe) and then used everywhere.